ord_schema/validations.py

```diff
@@ -326,6 +380,23 @@ def validate_addition_speed(message):
     del message  # Unused.
 
 
+def validate_crude_component(message):
+    if not message.reaction_id:
+        warnings.warn('CrudeComponents must specify a reaction_id',
+                      ValidationError)
+    if (message.HasField('has_derived_amount') and message.has_derived_amount
```

Comment:
It's a bit sloppy, but it should be safe to do `if message.has_derived_amount` since you'll never get `True` unless that was set explicilty.

Reply:
I think using `HasField` explicitly is a nice reminder that we have to be careful with optional Booleans, but this makes sense. I'll simplify for now, even though we might revert if we switch to using FrozenMessages